examples/esp32/esp32-server.rs

```diff
@@ -116,81 +162,61 @@ fn main() -> anyhow::Result<()> {
-
-    if !EthWait::new(eth.driver(), sl_stack)?
-        .wait_with_timeout(Duration::from_secs(30), || eth.is_started().unwrap())
-    {
-        bail!("couldn't start eth driver")
-    }
-
-    if !EspNetifWait::new::<EspNetif>(eth.netif(), sl_stack)?
-        .wait_with_timeout(Duration::from_secs(20), || {
-            eth.netif().get_ip_info().unwrap().ip != Ipv4Addr::new(0, 0, 0, 0)
-        })
-    {
-        bail!("didn't get an ip")
-    }
-    let ip_info = eth.netif().get_ip_info()?;
-    info!("ETH IP {:?}", ip_info);
-    Ok(eth)
+    eth: EspNetif,
+) -> anyhow::Result<(Ipv4Addr, AsyncNetif<EspNetif>)> {
+    use futures_lite::future::block_on;
+
+    let timer = EspTimerService::new()?;
+    let ip_info = eth.get_ip_info()?;
+    let async_eth = AsyncNetif::wrap(eth, sl_stack.clone(), timer);
+
+    block_on( async {
+        async_eth.ip_wait_while(
+                || Ok(async_eth.is_up().unwrap()), 
+                Some(Duration::from_secs(20)))
+            .await
+            .expect("ethernet couldn't connect");
+    });
+
+    info!("ETH IP {:?}", ip_info.ip);
+    Ok((ip_info.ip, async_eth))
 }
 
 #[cfg(not(feature = "qemu"))]
 fn start_wifi(
     modem: impl esp_idf_hal::peripheral::Peripheral<P = esp_idf_hal::modem::Modem> + 'static,
     sl_stack: EspSystemEventLoop,
-) -> anyhow::Result<Box<EspWifi<'static>>> {
-    use embedded_svc::wifi::{ClientConfiguration, Wifi};
-    use esp_idf_svc::wifi::WifiWait;
-
-    let mut wifi = Box::new(EspWifi::new(modem, sl_stack.clone(), None)?);
-
-    info!("scanning");
-    let aps = wifi.scan()?;
-    let foundap = aps.into_iter().find(|x| x.ssid == SSID);
-
-    let channel = if let Some(foundap) = foundap {
-        info!("{} channel is {}", "Viam", foundap.channel);
-        Some(foundap.channel)
-    } else {
-        None
-    };
-    let client_config = ClientConfiguration {
+) -> anyhow::Result<Box<AsyncWifi<EspWifi<'static>>>> {
+    use futures_lite::future::block_on;
+    use esp_idf_svc::wifi::config::{ScanConfig,ScanType};
+    use embedded_svc::wifi::{AuthMethod, ClientConfiguration, Configuration};
+
+    let timer = EspTimerService::new()?;
+    let nvs = esp_idf_svc::nvs::EspDefaultNvsPartition::take()?;
+    let mut wifi = AsyncWifi::wrap(EspWifi::new(modem, sl_stack.clone(), Some(nvs))?, sl_stack, timer)?;
+    let wifi_configuration: Configuration = Configuration::Client(ClientConfiguration {
         ssid: SSID.into(),
+        bssid: None,
+        auth_method: AuthMethod::WPA2Personal,
         password: PASS.into(),
-        channel,
-        ..Default::default()
-    };
-    wifi.set_configuration(&embedded_svc::wifi::Configuration::Client(client_config))?;
+        channel: None,
+    });
 
-    wifi.start()?;
+    wifi.set_configuration(&wifi_configuration)?;
 
-    if !WifiWait::new(&sl_stack)?
-        .wait_with_timeout(Duration::from_secs(20), || wifi.is_started().unwrap())
-    {
-        bail!("couldn't start wifi")
-    }
+    block_on( async {
```

Comment:
if they have a blocking version of the wifi driver why no use that? there are no upside for use to jump into an async context here